packages/teams-js/test/public/visualMedia.spec.ts

```diff
@@ -0,0 +1,517 @@
+  beforeEach(() => {
+    utils = new Utils();
+    utils.mockWindow.parent = undefined;
+    utils.messages = [];
+    GlobalVars.isFramelessWindow = false;
+  });
+  afterEach(() => {
+    app._uninitialize();
+    jest.clearAllMocks();
+    GlobalVars.isFramelessWindow = false;
+  });
+
+  describe('Testing isSupported', () => {
+    it('should not be supported before initialization', () => {
+      setUnitializedRuntime();
+      expect(() => visualMedia.image.isSupported()).toThrowError(new Error(errorLibraryNotInitialized));
+    });
+  });
+
+  describe('Testing HasPermisison API', () => {
+    it('should not allow hasPermission calls before initialization', () => {
+      return expect(() => visualMedia.hasPermission()).toThrowError(new Error(errorLibraryNotInitialized));
+    });
+
+    Object.values(FrameContexts).forEach((context) => {
+      if (allowedContexts.some((allowedContext) => allowedContext === context)) {
+        it(`should throw error when visualMedia is not supported in runtime config. context: ${context}`, async () => {
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: {} });
+          expect.assertions(1);
+          try {
+            visualMedia.hasPermission();
+          } catch (e) {
+            expect(e).toEqual(errorNotSupportedOnPlatform);
+          }
+        });
+
+        it(`visualMedia should throw error when permissions is not supported in runtime config. context: ${context}`, async () => {
+          await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({ apiVersion: 2, supports: { visualMedia: {} } });
+          expect.assertions(1);
+          try {
+            visualMedia.hasPermission();
+          } catch (e) {
+            expect(e).toEqual(errorNotSupportedOnPlatform);
+          }
+        });
+
+        it('hasPermission call in default version of platform support fails', async () => {
+          await utils.initializeWithContext(context);
+          expect.assertions(1);
+          utils.setClientSupportedSDKVersion(originalDefaultPlatformVersion);
+          try {
+            visualMedia.hasPermission();
+          } catch (e) {
+            expect(e).toEqual(errorNotSupportedOnPlatform);
+          }
+        });
+
+        it('hasPermission call with successful result', async () => {
+          await utils.initializeWithContext(context);
+          utils.setClientSupportedSDKVersion(minVersionForPermissionsAPIs);
+          utils.setRuntimeConfig({ apiVersion: 1, supports: { visualMedia: { image: {} }, permissions: {} } });
+
+          const promise = visualMedia.hasPermission();
+
+          const message = utils.findMessageByFunc('permissions.has');
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(1);
+          expect(message.args[0]).toEqual(DevicePermission.Media);
+
+          const callbackId = message.id;
+          utils.respondToFramelessMessage({
```

Comment:
I don't see any framed vs frameless unit tests in this file. Am I missing something?

Reply:
Talked offline, and thanks a lot to @shrshindeMSFT !! He explained what is framed and frameless in unit test. Because visualMedia API will work on iOS and android mobile devices for now, and we will not test framed in the unit test.